src/App.vue

```diff
@@ -0,0 +1,253 @@
+<script setup>
+import { ref, onMounted } from 'vue'
+
+const origins = ref([])
+const inputText = ref('')
+const placeholderURL = ref('https://hoppscotch.io')
+const errorMessage = ref('')
+
+const DEFAULT_ORIGIN_LIST = ["https://example.com", "https://another-example.com"]
+
+const getOriginList = () =>
+  new Promise((resolve) => {
+    chrome.storage.sync.get(["originList"], async (items) => {
+      if (!items || !items.originList) {
+        await storeOriginList(DEFAULT_ORIGIN_LIST)
+        resolve(DEFAULT_ORIGIN_LIST)
+      } else {
+        resolve(JSON.parse(items.originList))
+      }
+    })
+  })
+
+const storeOriginList = (originList) =>
+  new Promise((resolve) => {
+    chrome.storage.sync.set(
+      {
+        originList: JSON.stringify(originList),
+      },
+      () => {
+        resolve()
+      }
+    )
+  })
+
+const onAddClick = async (event) => {
+  event.preventDefault()
+  try {
+    const parsedURL = new URL(inputText.value)
+    if (origins.value.includes(parsedURL.origin)) {
+      errorMessage.value = "Origin is already on the list"
+    } else {
+      origins.value.push(parsedURL.origin)
+      inputText.value = ""
+      await storeOriginList(origins.value)
+      errorMessage.value = ""
+    }
+  } catch (e) {
+    errorMessage.value = "Improper URL"
+  }
+}
+
+const onInputTextChange = (ev) => {
+  inputText.value = ev.target.value
+  errorMessage.value = ""
+}
+
+const onDeleteOriginClicked = async (index) => {
+  origins.value.splice(index, 1)
+  await storeOriginList(origins.value)
+}
+
+onMounted(() => {
+  getOriginList()
+    .then((list) => {
+      origins.value = list
+    })
+    .catch(() => {
+      // Just fail silently :P
+    })
+
+  chrome.tabs.query({ active: true }, (result) => {
+    if (result.length > 0) {
+      try {
+        if (result[0].url && result[0].url.startsWith("http")) {
+          const url = new URL(result[0].url)
+          if (url && url.origin) {
+            placeholderURL.value = url.origin
```

Comment:
here inputText.value will override the placeholder. placeholder will only be visible when inputText is empty. so there's no need to make placeholderURL a ref, it can be a simple variable.

also you can remove this assignment

```suggestion
```

Reply:
Resolved, I don't know why I did this blunder, will raise a commit on this PR soon.